src/type/introspection.js

```diff
@@ -108,8 +109,7 @@ var __Type = new GraphQLObjectType({
         includeDeprecated: { type: GraphQLBoolean, defaultValue: false }
       },
       resolve(type, {includeDeprecated}) {
-        if (type instanceof GraphQLObjectType ||
-            type instanceof GraphQLInterfaceType) {
+        if (typeHasFields(type)) {
```

Comment:
Misread the previous, but this is still probably not safe as now InputObject is going to present fields as the wrong type and in the wrong place in introspection